src/panels/config/devices/ha-config-devices-dashboard.ts

```diff
@@ -338,11 +338,24 @@ export class HaConfigDeviceDashboard extends LitElement {
               : undefined;
           const batteryIsBinary =
             battery && computeStateDomain(battery) === "binary_sensor";
+
+          let battery_state_display = battery?.state;
+          if (
+            typeof battery?.state === "string" &&
+            battery.state.trim() &&
+            !isNaN(Number(battery.state))
+          ) {
+            // Remove trailing 0s from battery decimal
+            battery_state_display = Number(battery.state).toString();
+          }
+
           return battery && (batteryIsBinary || !isNaN(battery.state as any))
```

Comment:
This seems like a (maybe harmless) bug.  The whole expression should be surrounded by parentheses.

Reply:
which expression?

Reply:
Everything after `return` before the `?` in the ternary.  Without the parentheses, the ternary only applies after the `&&` so it returns `battery` when `battery` is falsey.